src/gtest/test_keystore.cpp

```diff
@@ -359,15 +359,24 @@ TEST(KeystoreTests, StoreAndRetrieveMnemonicSeedInEncryptedStore) {
     seedOut = keyStore.GetMnemonicSeed();
     EXPECT_FALSE(seedOut.has_value());
 
-    // Unlocking with a random key should fail
-    CKeyingMaterial vRandomKey(32, 0);
-    GetRandBytes(vRandomKey.data(), 32);
-    EXPECT_FALSE(keyStore.Unlock(vRandomKey));
-
-    // Unlocking with a slightly-modified vMasterKey should fail
-    CKeyingMaterial vModifiedKey(vMasterKey);
-    vModifiedKey[0] += 1;
-    EXPECT_FALSE(keyStore.Unlock(vModifiedKey));
+    // Unlocking with a random key causes sporadic failures, since we currently
+    // don't use an authenticated encryption scheme for CCryptoKeyStore.
+
+    // Currently, DecryptMnemonicSeed tests if a key is invalid by looking at
+    // the return value of CBCDecrypt. If keyStore.Unlock is called with an
+    // invalid key, there's roughly a 257/65536 chance that the padding check
+    // in CBCDecrypt will pass, in which case DecryptMnemonicSeed then calls
+    // the deserialization code in mnemonic.h with random data.
+
+    // // Unlocking with a random key should fail
+    // CKeyingMaterial vRandomKey(32, 0);
+    // GetRandBytes(vRandomKey.data(), 32);
+    // EXPECT_FALSE(keyStore.Unlock(vRandomKey));
+
+    // // Unlocking with a slightly-modified vMasterKey should fail
+    // CKeyingMaterial vModifiedKey(vMasterKey);
+    // vModifiedKey[0] += 1;
+    // EXPECT_FALSE(keyStore.Unlock(vModifiedKey));
```

Comment:
```suggestion
```
There's no need to retain this code; we can find it in the history if and when we switch to properly authenticated encryption.

Reply:
SGTM, force-pushed with that.